blog/content/posts/2025-12-xx-transport-direction.md

```diff
@@ -0,0 +1,83 @@
+---
+date: "2025-12-09T09:00:00+00:00"
+publishDate: "2025-12-09T09:00:00+00:00"
+title: "Transport Direction"
+author: "Kurtis van Gent, Shaun Smith"
+tags: ["mcp", "governance"]
+ShowToc: true
+---
+
+# MCP Transport Roadmap
+
+When MCP launched in November 2024, the most common way to use it was locally - connecting Clients to Servers over STDIO. As MCP has grown to become the standard choice for LLM integration, the need to deploy MCP in distributed environments - and at scale has grown.
+
+Pioneers of remote, scaled deployments using the Streamable HTTP transport have found practical challenges which make reusing existing infrastructure deployment patterns hard.
+
+## Roadmap
+
+To address these challenges, the Transport Working Group has spent the last few months collecting feedback and ideas from the community. We have worked closely with the Core Maintainers to propose solutions to the most common problems.
+
+In this post we share the roadmap for evolving the Streamable HTTP transport, and invite community engagement to shape the future of the protocol.
+
+### A Stateless Protocol
+
+At the outset, MCP was designed to be _stateful_, with Clients and Servers maintaining knowlege of each other over a bidirectional communication channel.
+
+Connections are initialized with a handshake which shares information like Capabilities and Protocol Version. This state remains fixed for the duration of the connection, and requires techniques such as sticky sessions or distributed session storage to for scaled deployment.
+
+We propose making MCP stateless by:
+
+- Replacing the `initialize` handshake - and sending the shared information with each request/response instead.
+- Providing a `discovery` mechanism for Clients to query Server Capabilities if they need the information early (for example for UX purposes).
+
+We're moving toward a more dynamic model where clients can optimistically "just try" what they need and receive helpful error messages if unsupported.
+
+> [!NOTE]
```

Comment:
@evalstate @kurtisvg would recommend building this with Hugo locally - I am not sure if the GitHub note shortcode will render on the blog itself.